src/shared/components/views/DatatableWithToggle.tsx

```diff
@@ -20,29 +25,37 @@ const DatatableWithToggle = ({ table }: { table: JSX.Element }) => {
     'protvista-datatable'
   );
 
-  const toggleTableState = async () => {
-    const tableEl = tableRef?.current;
-    if (tableEl) {
-      if (tableState === 'Expand') {
-        tableEl.expandTable = true;
-        setTableState('Collapse');
-      } else {
-        tableEl.expandTable = false;
-        setTableState('Expand');
-        await frame();
-        tableContainerRef?.current?.scrollIntoView({
+  // On expand/collapse change
+  useEffect(() => {
+    // except on first render
+    if (!firstRenderRef.current) {
+      // Scroll table back into view when collapsing
+      if (!expandTable) {
+        tableRef.current?.parentElement?.scrollIntoView({
           behavior: 'smooth',
           block: 'center',
         });
       }
+      gtagFn('event', 'feature viewer', {
```

Comment:
Took the opportunity to add some tracking as it is a behaviour we might be interested to see in the analytics numbers